e2e/test/scenarios/admin/i18n/content-translation/column-names-in-viz.cy.spec.ts

```diff
@@ -0,0 +1,289 @@
+    const columnY = "RATING";
+
+    const priceVsRating: Pick<
+      StructuredQuestionDetails,
+      "query" | "visualization_settings"
+    > = {
+      query: {
+        "source-table": PRODUCTS_ID,
+        filter: [
+          "and",
+          ["between", ["field", PRODUCTS.PRICE, null], 40, 60],
+          ["between", ["field", PRODUCTS.RATING, null], 3.7, 4],
+        ],
+      },
+      visualization_settings: {
+        "graph.dimensions": [columnX],
+        "graph.metrics": [columnY],
+      },
+    };
+
+    const priceVsAverageRating: Pick<
+      StructuredQuestionDetails,
+      "query" | "visualization_settings"
+    > = {
+      query: {
+        "source-table": PRODUCTS_ID,
+        aggregation: [["avg", ["field", PRODUCTS.PRICE, null]]],
+        breakout: [["field", PRODUCTS.RATING, null]],
+      },
+      visualization_settings: {
+        "graph.dimensions": ["PRICE"],
+        "graph.metrics": ["RATING"],
+        "graph.y_axis.title_text": "Average Price",
+      },
+    };
+
+    const visitEmbeddedViz = (questionDetails: StructuredQuestionDetails) => {
+      questionDetails.enable_embedding = true;
+      questionDetails.name = questionDetails.display;
+      cy.signInAsAdmin();
+      return H.createQuestion(questionDetails, {
+        visitQuestion: true,
+      }).then(() => {
+        if (questionDetails.display === "pivot") {
+          H.saveSavedQuestion();
+        }
+        H.openStaticEmbeddingModal({
+          acceptTerms: false,
+          activeTab: "parameters",
+        });
+        return H.getIframeUrl().then((iframeUrl) => {
+          cy.signOut();
+          cy.visit(`${iframeUrl}&locale=de`);
+        });
+      });
+    };
+
+    const assertColumnNamesAreTranslated = () =>
+      columnsInChart.forEach((row) => {
+        cy.findByTestId("visualization-root")
+          .get("text")
+          .should("contain", row.msgstr)
+          .should("not.contain", row.msgid);
+      });
+
+    const columnsInChart = germanFieldNames.filter((row) =>
+      [columnX, columnY].includes(row.msgid),
+    );
+
+    /** To ensure that all display types are handled, each test pushes a
+     * display type to this array. At the end, we make sure that this array
+     * includes all the display types in the CardDisplayType type.
+
+      We initialize it with a few display types that do not have localizable
+      column names. */
+    const tested: CardDisplayType[] = [
+      "scalar",
+      "smartscalar",
+      "gauge",
+      "progress",
+
+      // The names in a sankey query come from the SQL and are not
+      // currently localizable
+      "sankey",
+
+      // This isn't a graphical visualization so it's not tested here
+      "table",
+    ];
+
+    it("bar", () => {
+      visitEmbeddedViz({
+        ...priceVsAverageRating,
+        display: "bar",
+      });
+      assertColumnNamesAreTranslated();
+      H.chartPathWithFillColor("#A989C5").eq(3).trigger("mousemove");
```

Comment:
I notice in the pie chart test we use realHover. Did that not work here?

Reply:
I probably I tried that first, and for whatever reason, I had to use this pattern instead to make the tooltip appear in this test